modules/serverpod_auth/serverpod_auth_server/lib/src/business/user_authentication.dart

```diff
@@ -62,7 +62,7 @@ class UserAuthentication {
     Session session, {
     int? userId,
   }) async {
-    userId ??= (await session.authenticated)?.userId;
+    userId ??= (session.authenticated)?.userId;
```

Comment:
Remove parenthases around fetching authenticated info. 

```suggestion
    userId ??= session.authenticated?.userId;
```

Reflect this change in all other places that access this field.